lib/spack/spack/test/spec_list.py

```diff
@@ -9,11 +9,11 @@
 
 class TestSpecList(object):
     default_input = ['mpileaks', '$mpis',
-                     {'matrix': [['hypre'], ['$gccs', '$clangs']]},
+                     {'matrix': [['hypre'], ['$%gccs', '$%clangs']]},
```

Comment:
I'm explicitly trying to avoid parsing text as part of the tests for SpecList, because the SpecList object does not actually parse a yaml file. It is given inputs from a yaml file by the Environment object that uses it. So I think yaml parsing should only happen in Environment tests, not SpecList tests.

I will create a separate test.